### ISSUE TITLE
Add GeoCentroid as an aggregation

### ISSUE THREAD

**Issue by coreation** [ruflin/Elastica#1145]
Hello @ruflin ,

I've been using this excellent library for a while now, loving it!
Now I need to use a centroid aggregation on a geo_point field, however I don't see to find this anywhere in the repository. Is it a hidden feature somewhere, can I use a centroid aggregation in a different way or should I try to implement it?

**Comment by ruflin** [comment]
If I understand you right, this is what you are looking for on the elasticsearch side: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-geocentroid-aggregation.html

This currently does not exist yet in Elastica. It would be great to have this added as a feature.

**Comment by ruflin** [comment]
@coreation As a shortcut you can always use raw queries (http://elastica.io/example/raw-array-query.html) but it would be obviously much nicer to have it directly in Elastica.

**Comment by coreation** [comment]
Cool, if I understand it correctly I can use this for example in a geohash grid aggregation with a geocentroid sub aggregation to make the center of that grid an average of all locations in that grid right? If so, I might be very tempted to spend some time making this into a PR :).

**Comment by ruflin** [comment]
Yes. The "only" thing you have to do is adding support for geocentroid aggregation. As each aggregation inherits AbstractAggregation, you will get the nesting part for free.

**Comment by coreation** [comment]
I already looked into that before I made this issue, just to make sure I wasn't missing anything. I'll probably try this tonight and ask for feedback on my fork before making a PR.

**Comment by ruflin** [comment]
Best is to open a PR directly. I see PR's as a type of conversation and not something that is final and must be directly ready to be merged. Code often says more then lots of descriptions.

**Comment by coreation** [comment]
Alright, will do!

**Comment by coreation** [comment]
@ruflin I think I got it, if so that was by far the easiest PR and open source contribution in my life so far ;). I'll make a PR, but I'll have to adjust it either way, not sure where I should add this feature in the Changelog.md

**Comment by ruflin** [comment]
Thanks for the PR. Best is to add it under `Added` in the CHANGELOG.md

**Comment by ruflin** [comment]
`Elastica\Aggregation\GeoCentroid` exists and can be used directly:

```php
$agg = new \Elastica\Aggregation\GeoCentroid('centroid', 'location');

$query = new \Elastica\Query();
$query->addAggregation($agg);
```

Closing as implemented. Thanks for the kind words about the library.
